nixos/modules/services/games/minecraft-server.nix

```diff
@@ -61,13 +143,26 @@ in
       serviceConfig.User    = "minecraft";
       script = ''
         cd ${cfg.dataDir}
+
+        ln -sf ${eulaFile} eula.txt
+        ${if cfg.whitelist != null then "ln -sf ${whitelistFile} whitelist.json" else ""}
+
+        # create backup of original server.properties
+        if [ -f "server.properties" ]; then
+            cp -n server.properties server.properties.old
+        fi
+        cp ${serverPropertiesFile} server.properties
```

Comment:
You can just use `cp -b ${serverPropertiesFile} server.properties`. The `-b` automatically backups the old file if it exists.

Reply:
Also: Wouldn't it be safer to just not copy it if there's a file present already? This would automatically allow users to use a stateful properties file. Not everybody wants to transfer their server.properties file to NixOS just to have it declarative. Also: This would silently change people's server configs without warning. So I'm definitely against removing any other properties files.

Reply:
I've heard that several times. So I think the consensus is never overwrite configuration files. So `cp -n` should be used?

Reply:
If it works yeah, but I've seen a lot of `if [ ! -f file ]; then cp ...` in nixpkgs